### PR TITLE
Typings: use Config type

### PR DESCRIPTION
**Pull request by danmarshall**
I believe this is the intention, perhaps it was set to `any` for a reason?

### CODE REVIEW

File: packages/vega-typings/types/spec/legend.d.ts [hunk 1]
```diff
@@ -302,6 +302,14 @@ export interface BaseLegend<
    */
   gradientStrokeWidth?: N;
 
+  /**
```
Comment (domoritz) [inline packages/vega-typings/types/spec/legend.d.ts]: Let's not add a deprecated property to the typings.

File: packages/vega-typings/types/spec/config.d.ts [hunk 2]
```diff
@@ -415,8 +415,8 @@ export interface BaseLegendLayout<
   BS = boolean | SignalRef,
   OS = Orientation | SignalRef,
   LB = LayoutBounds,
-  AN = TitleAnchor
-> {
+  AN = TitleAnchor | SignalRef
```
Comment (domoritz) [inline packages/vega-typings/types/spec/config.d.ts]: Can you provide a link to the Vega schema or docs that shows that this is valid?

Comment (danmarshall) [reply packages/vega-typings/types/spec/config.d.ts]: The [Config / legends / layout](https://vega.github.io/vega/docs/config/#legends-layout) doc does not specifically say you can or cannot use a signal. However, the [Signlas doc](https://vega.github.io/vega/docs/signals/) states:
> Signals can be used throughout a Vega specification

In practice, the signal is functional for legend layout:
https://vega.github.io/editor/#/url/vega/N4KABGBEAkDODGALApgWwIaQFxUQFzwAdYsB6UgN2QHN0A6agSz0QFcAjOxge1IRQyUa6SgFY6AK1jcAdpAA04KIXQATVYxnVsYAIwAGRREjxZAM0bacoCMfQAPRrB03bxvI3gBrAMLcANtwATjqQAMTwkQpKbpCq3BiafoEhOOGR8JAxYAC+RraQ-jTIMqou2cb+6ACe3Kx45W6x6DJIwS6QsJYy6P6hRdQlqgCCrYjtORW52ZMQeSBKnd29zjgA2tnAUD2oyKFI6EF4AMqMAF578lAUvax7OLoATPq5+RBbkDv3UAdHAApqBTXW7fADML3mbg+X32iEOJzwyEIQMgrEIqnQiNh8NOFzAAGowL88ACyq9NlNPuhdvtkP4+ldIDd-HcdDJWPS3gVZDoNk1bK5+cZkFQZHhVlAAALwOn+LCoOqwZDxADuckZaIxWLS1CCdUIAAoAJRZIXks2CoWQEUlcWhaWy+WK5Bo6Jm4yazHfSAAQlgKhkBvYQVYsEQAA0jWAAGTRsB+gNBkNhgCaUYA-GB2fSwDgZfTTULZk0ALozLmW4wwtLB0MRlHM1k4QyUnnrKbvDsFG1iiWQB30p2h5XcNVu92o9Fe0JrewG-P+I1XOcLo0lwv8yFCytNa2iu1pNYDuUK4eqmRXFWaVVDpVoktgAB8YCvpVHt7Q3Co47Nk613rWWsw3DNZ9BLK54CqVBDRXWUlzAQwiThI5cWQNcNyaLd+R3WIewPMBgCWagej6NJVDpPBMCwq1PW1H4oMIAAlFpBgNNYWjaIIQLAgkwHI-xKNA8CwA48YuLWXQH0JfjBLA4TEOJVCTS7Ys3DLNxqJ3KkaRrZNEBTBsQR0Ft+UgNswD5bcu2Ffc+2PD9zx-GipzoyA1mqec4KuDzV3XFSuShayoDwvsjwXBzRwvF9r3fU870IB9n1fG84s-b8At3WiAKA-ShIghiDR8ryEIg5CTnONC-LNajAvdYLbI6LpiN6UIZKojLYiy-YGOYrRkDY0TghTITeLaiThMGoJhsk0aKPQcb5NKnEKuU6qpnU2xNMpasoEcxlG2+UCrjAjKzLkdZCJC+0coiscNRcgDYPpeCiue9ccg2uYK226lvUm8NDJZb5s38U7zLWS6GrSSUbtSvaoC6tJYH8Tx+pyyNIA+8sKVMnbIEmgz9qMnAQbB86LMh207Nh514b-ackZRmUkzrNNMc+80BR+nTdrmwGmwsxCTtbcnLOwoK9ypmcYb026ouS2LnXvJ9orfFUPwVdKJcRqBZ0856wAAWj4yK8rAV7FyNk21QWjCNPW7Hat3PGghY5AAaJoHeSF8CRd5LscO7KGCKIkjQnRzGOoKHXIE0KgjgNAByTQZGQLjE6udGILgu3NodjTvtx37QldvrCeBL31h9sn-YtCWrusUOWt01nI+1h7QjjtO8CTlO05TDOwBylNs+e3O5nzzbC+d4u0ngVggmkEJPYFyBE-Isx0A5PBE6c4xwYD+vg7c+y4ciy8YvV1LlaSy+Na-ZAfT3zKO7SROZBVJVDaCZAuguXf-KHzqpLXs11ZapU1sgK4R4aZnnPqrFKSsEoqwVlfZ0kCn5Rw9K-KAidIEAOqlHQONkpbQ1gUqOoeAL5qw-K6LBCMcFrw3lvASBCiyTy+jjGePNCg1EoQAITqKUZwK9vRmBZGGZ+kB2DXg6JoQg9RQhKiKPABojJuCEA8LICUaxIDiLrCiMwHI+gllUpzTsRceFVFqPUYYKNiL8z+gWU6MjShyJkAohoSM6TIFUSiDRWiZA6Pxs44K6AkAohkLIPYpjHZc0sd6axlDGKjlJI4nQBgXGyMbvIxRaRS6DBRKgTQxlGQYHsDoZ4jJYCImRA8MxW0En9D4fUZIaTREZJMruVxZQckeLyVAAplwoDFPJohSA5TKnjJqUiDJDTp6xDxkk+oAB5MwZglRqIrgLLpsQenuM8SXN2RSSnNjKQ4Tp1TalzLiRY7hiTiilFGJxdJSNKJHCkfsvphzvEqK2WZTRPAgm8k6O8-5xT1BFBREMSAsT7YQF9osLUmB2xO0WbPKAjAghOCkQvUiu1MQiCxU4SQ0g5C3IItzMRjA6SqBEZSA6wTCDIEov4AA6owVQLAUTMtZQAGRKNQbl1SkS9A5VyxAKIlQqH8AKrQ3KOaNPufsPUsB6WmWkAvGUoQLC0vVbuPApdYBmGCKgWu24oB4GqMylV3A1UonQMEipjJqiwvMU0D4VqbVpBNUEVAHJMCMkdaEewCBejDOtPYQgqQCV4FYKgOg9g6DIt4oncMu93VQktdasRpqA0Or7NUMNULGTICjTGuImJ410GqMmwlqaB6QGmPyRVShEXGGLb-c1dz0U8NMCkKRXrvTBA0GHU68REjk0Isizu2KRFQF1f4XpUB-S+Jpc4Gqgzjlzy9NQYIrqZhtsUIsAYQwdFcNiBYAsc8AjtFOh4PAUK0jHGZfAddUjuDrM2VM06JRTDkUaFaWA1RUDsACBKYhDD-yAeAVe-wKyVBvqtR0A6xk6CiE3Rqw13AvDfEIqhtIhqWj+kOLaTGQUzHwswoehFx7KgtK8ZS2IKh1CaCsEx3ceoVSNWWPi3hNi8ApJVGkzdJgAjxqnU3Pjyy8BtMBPM7IZkv0sp481aTDG1kbJU9RMTLJUDApwAAFi5NIoRdLVNhzSDJwRrBhFt2aPYyTTVLNQBk3Y7o5HOGLAwEELw560UFCHSXXxajKR-u4AB6wXZrRijTjBq0cGUPEygGEZAaXPNFijvTOikGAXhOYK6xuznm4-FlGATMuhcwIXsxOCpRXeO0hzJmBcibeLoxGs2Grv57CPAsyVkwZXmuyla4SdrM1OuYYKIVkOxW+MLnK0SYb1Q2t6WGjxCb9DIDVF6-VtTjX-ALZa8t0bq3xpVf0BlzcUwFMBarBigbzjKRBZ1HqOhlIzB6jNY3Gdc9VUbtOnqJ9K7TA2ro7ucLkWOOmVtHFqLwCrwSr63NsqqEutWhQJYfASPsQoQqmj0yiXG4EctUalQP8xT493DUvUuGkuVxS+oMolG86bjB7EWb-ntzaW9Aj4VUGGY-BR3jybhE8YY+oFj+60G55C4uJd2w7aqfwHDZz8WwC8Z9y4iiZ7rnNDIEOM-WIFw9Q6E3v4JUjIZCox0Iau4m2J3oFOSHH7mK52GJpUu7HaQye2kTXWyimNJvGCGaEXnkrAF13V-dzX5dIA68KHrg3m3jfcFNysKB2xrc4Ft1AiWDunfTsJbOnFjJF3LsIrN0IPuxQ1v91RIPW6+qhHF1jlS602cFB8357tHqJbx+A6BgIhuCgfYSB0F3sc3cN9jrF12qjGDfhwGbi39dWgRbw0FYUs-4vALq1DuqBRO2dxkKnLX9CCce-L5WuNX2Q7V8Y5AJNM6cjM8IZvqbO+D+dGV0Dmfp-Cbv6XqX7j5Vq36ET36hC1rP6v6ZaAELqMD0gIb5bIZE7Jb6DoYN7s546oH06ggABsMBBc7+2WG+B+I+CB+KYsZB++1Blqv8D+Po82AAPkwWAJoEMgaFqPGmsC1lwQmk-oSsJEMpGDGHGOwW7JwaATwcNnwbXsikIW7GzOfkBj-t6P2nevAbSkoq+u+nAZwrQe8EyMluEIzvLnVBzGtHVDAczoqgsgUIDt6FxobCgGoHFvejmqELqPqFIqPmAdfiipoUuv9pSJoIiHPh4IvmAMvrnmFmvpDoHDFmEZ-rEGHnTjstPi3oxhXg1jLstHLtPoTiHMTnHqTqRhTtYddh3h2rxqrh6tzs3sgJjv8jHIpMLu3gsBqqobUbdsqmkDHtrh4VZonsvBLCnmnubhnp8FnmADnvbgkI7pJhPsSvOnosAbti5pAPfnXoHptiHmkJkePM2qWFUVAA4F2qilzsEDShTlZsgGYP8kfn0Sfv3Nrp4L4EIoxqIFgnHswEDtkXtt7mUXgNsS-hwu6lpA4SXKOIbCatwEke4d6lAF4W9gTp9iAQHj6pfsEaZKEWnOEhEd8NEScTFv+qQVaDDjGrlqkTgekZtgcesf1q0fkZtoUfhsYURkEhARUazh0VTjUT3lCPUfsY0RLs0YwkyXsDYcSZ2t0fEhOBrs8VNAMYiQnqnEnqMWnKnkvunpbtMbMXnvMQXv4cXisWXl7soECSCdPnsbgCKa3pYccbyc0PYOcRZAHGZNimRvkk0VKqocfv-q8d4H4LZp8d8Q+n8VJlXpadAWCUqrEJCTenpjIM4fruRCMaZPHsiciKdL4eiQEasXqpTjPmEfiQvoSTqU6d2HEWSdDtvnDhONSTNjkYLnkZKXSXaVkUYfTvoAURQWkd6BySRuTg0NyZhNKfyRcdhEKVAKkVLgLiYLLpKe0ZSDKQKXKb+Aqafh7NmiqSjGqemROGMdqRMbqdqtniGLnsAvnosUXn0W7qXmsU2QCRacOVabsdujOZytym3i2sSWcbKYYVcV6Zahor6eGv6WnNuT8UGR8ToF8dZD8Y+nhpGYCa+TGS2hShCQEGoeJvpjCdwHCW4U9oMUia9tme9mid9reYEeZqJrieEWWeMSvrEaSckcFHWTQbuI2f8RsRKZTrEPSUUWgb2dekJfTiUcRlyT+WOZWdUWpgBSHHjLOfzq5HxVKbJcDirmuT2laJuZBcqYksMcPsYEeVETqZnmeTMReXMZOnmSae7loQyXxlsdAe+U3mkKkdJWpH+S6QpYREBTcVAGBgQGPtUn6U8VuYGe8SGXBWGb8chZXqhb7uhaWJhcRSqT-H4qdHjDlFIhDjWeDhxblqyV2avIOVJW-r+DrLlnvjxYyYNotvSCNkPHpNxA+BthLD1uaQ9gdkNk1fYCtnWCBONtVtPtNnVXNg1UdoNamB1qNZtttt1fNn1f4DWjNblCNRdqOVRhPAiiADkEAA

Comment (domoritz) [reply packages/vega-typings/types/spec/config.d.ts]: Thanks. It turns out that some properties don't support signals so we always need to be careful.

Comment (domoritz) [reply packages/vega-typings/types/spec/config.d.ts]: Can you check the Vega schema as well?

**Comment by domoritz** [comment]
I replaced this with separate PRs. https://github.com/vega/vega/pull/1934

**Comment by domoritz** [comment]
Please open a new PR for the titleAnchor. In particular, please link to the schema and also look at whether we need to change the typings for axes and headers as well.

**Comment by danmarshall** [comment]
See #1935

**Pull request closed without merging**